File: 0034-find-first-and-last-position-of-element-in-sorted-array/solution.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] output =new int[2];
        output[0]=-1;
        output[1]=-1;
        int c=0;
        for(int i=0;i<nums.length;i++){
           
                if(nums[i]==target){
                    output[0]=i;
                    break;

                }

            
        }
        if(output[0]!=-1){
            for(int i=nums.length-1;i>=0;i--){
                if(nums[i]==target){
                    output[1]=i;
                    break;
                }
            }
        }
        return output;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/solution.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int n = nums.size();
        int start = -1, end = -1;

        // Find the first occurrence (start)
        int l = 0, r = n - 1;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if (nums[mid] == target) {
                start = mid;
                r = mid -
                    1; // Keep searching to the left for the first occurrence
            } else if (nums[mid] < target) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }

        // If the target wasn't found at all, return [-1, -1]
        if (start == -1) {
            return {start, end};
        }

        // Find the last occurrence (end)
        l = 0, r = n - 1;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if (nums[mid] == target) {
                end = mid;
                l = mid +
                    1; // Keep searching to the right for the last occurrence
            } else if (nums[mid] < target) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }

        return {start, end};
    }
};

